src/nopoll_helpers.c

```diff
@@ -49,59 +50,32 @@ void setMessageHandlers()
     nopoll_conn_set_on_close(get_global_conn(), (noPollOnCloseHandler)listenerOnCloseMessage, NULL);
 }
 
+static int cloud_status_is_online (void)
+{
+	const char *status = get_parodus_cfg()->cloud_status;
+	if (NULL == status)
+	  return false;
+	return (strcmp (status, CLOUD_STATUS_ONLINE) == 0);
+}
+
 /** To send upstream msgs to server ***/
 
 void sendMessage(noPollConn *conn, void *msg, size_t len)
 {
     int bytesWritten = 0;
-    static int connErr=0;
-    long timeDiff = 0;
+
+    if (!cloud_status_is_online ()) {
+		ParodusError ("Cloud Status not ONLINE at nopoll send\n");
```

Comment:
We should retain these logs:
ParodusError("Failed to send msg upstream as connection is not OK\n");	        
OnboardLog("Failed to send msg upstream as connection is not OK\n");